test/pylib/log_browsing.py

```diff
@@ -0,0 +1,95 @@
+# Copyright 2023-present ScyllaDB
+#
+# SPDX-License-Identifier: AGPL-3.0-or-later
+
+import asyncio
+from concurrent.futures import ThreadPoolExecutor
+import logging
+from typing import Optional
+import pytest
+import os
+import re
+
+
+logger = logging.getLogger(__name__)
+
+class ScyllaLogFile():
+    """
+    Class for browsing a Scylla log file.
+    Based on scylla-ccm implementation of log browsing.
+    """
+    def __init__(self, thread_pool: ThreadPoolExecutor, logfile_path: str):
+        self.thread_pool = thread_pool # used for asynchronous IO operations
+        self.file = logfile_path
+        if not os.path.isfile(self.file):
+            pytest.fail("Log file {} does not exist".format(self.file))
+
+    async def _run_in_executor(self, func, *args, loop=None):
```

Comment:
I don't get why you added the `loop` parameter. Isn't it always obtained through `get_running_loop()`? Why sometimes obtain it outside the function and sometimes inside?

Reply:
I wanted to reuse `loop` inside `wait_for` and `grep` to not call `get_running_loop()` on each read line from the log file.
I also wanted to make calling it easier in case when it called only once.

Reply:
Ok